File: code/basics/type-coercion.erl
% Type coercion in Erlang
% Erlang doesn't have implicit type coercion
% Conversion must be explicit
-module(example).
-export([start/0]).

start() ->
    IntNum = 42,
    FloatNum = 3.14,
    % Convert to string
    StrInt = integer_to_list(IntNum),
    StrFloat = float_to_list(FloatNum),
    % Convert from string
    IntVal = list_to_integer(StrInt),
    FloatVal = list_to_float(StrFloat),
    io:format("~p ~p~n", [StrInt, StrFloat]).

File: code/basics/type-coercion.ex
# Type coercion in Elixir
# Elixir doesn't have implicit type coercion
# Conversion must be explicit
int_num = 42
float_num = 3.14

# Convert integer to float
float_val = int_num / 1
# Convert float to integer
int_val = trunc(float_num)
# Convert to string
str = Integer.to_string(int_num)
str = Float.to_string(float_num)
# Convert from string
int_val = String.to_integer("42")
float_val = String.to_float("3.14")